sdk/core/azure-core/tests/test_pipeline.py

```diff
@@ -157,5 +145,119 @@ def test_request_url_with_params(self):
         self.assertIn(request.url, ["a/b/c?g=h&t=y", "a/b/c?t=y&g=h"])
 
 
+class MyPipeline:
+    def __init__(self, policies=None):
+        self.request = HttpRequest("GET", "https://bing.com")
+        if policies is None:
+            policies = []
+        self.transport = RequestsTransport()
+        self.pipeline = Pipeline(self.transport, policies=policies)
+
+    @use_distributed_traces
+    def run(self, numbTimes, **kwargs):
+        if numbTimes < 1:
+            return None
+        response = self.pipeline.run(self.request, **kwargs)
+        self.run(numbTimes - 1, **kwargs)
+        return response
```

Comment:
I still think `run` is not the method where `@use_distributed_traces` will be applied, the main usage will be to decorate client functions that run pipelines:
```
class TestClient:
 
    @use_distributed_traces
    def get_foo(self):
```
So we should test that. A basic unit test of `use_distributed_traces` can now just wrap an arbitrary function, and check that the context is updated appropriately. A more comprehensive "end to end" test would need to create a fake `TestClient` that:
* configures a pipeline with a tracing policy
* configures a "transport" that records the request  that is sent out (for testing comparison)
* wraps a `get_foo` methods with the decorator.

Then calling `client.get_foo` the request can be inspected to see that the appropriate headers have been inserted. I think I've seen some similar tests in some of @chlowell PRs

Reply:
I consulted his PRs and made some changes. Is this the type of stuff you were talking about?